### PR TITLE
Add `from_od` flag to the `RemoteNode.load_configuration`

### PR DESCRIPTION
**Pull request by vongostev**
In the library we have some methods to load settings from OD, for example `node.rpdo.read(from_od=True)`, but we can not do this for entire configuration.
This PR fixes it.

### CODE REVIEW

**Comment by acolomb** [comment]
What exactly is the use case for this?  `load_configuration()` actually does not upload the objects *from* the remote node, but it downloads the OD variable values *to* the remote node.  Then it goes on to read back the applied PDO configuration.

How does it make sense to buy in to the vast exchange of SDO downloads, but then saving a couple of SDO uploads for reading back the actually accepted configuration?

File: canopen/node/remote.py [hunk 1]
```diff
@@ -154,4 +154,4 @@ def load_configuration(self):
                         self.__load_configuration_helper(subobj.index, subobj.subindex, subobj.name, subobj.value)
             elif isinstance(obj, ODVariable) and obj.writable and (obj.value is not None):
                 self.__load_configuration_helper(obj.index, None, obj.name, obj.value)
-        self.pdo.read()  # reads the new configuration from the driver
+        self.pdo.read(from_od=from_od)  # reads the new configuration from the driver
```
Comment (sveinse) [inline canopen/node/remote.py:157]: This seems to be the reason for adding the `from_od` argument. The current imeplementations calls `self.pdo.read()` with default parameters will read from the remote.

Maybe it's better to do `self.pdo.read(True)` and not add the `from_od` argument? The function description does not allude to it doing something against the remote.

**Comment by sveinse** [comment]
Perhaps its better to just call `self.pdo.read(True)` and skip the `from_od` argument to the function?

**Comment by acolomb** [comment]
> Perhaps its better to just call `self.pdo.read(True)` and skip the `from_od` argument to the function?

The `.load_configuration()` method is more equivalent to `pdo.write()`, though.  It goes through all objects in the OD actually and downloads the values to the remote node via SDO.  Then to avoid having mismatching information about the PDO-related objects, it reads back (uploads) the PDO records again via SDO.

In essence, there is nothing to fix here IMHO, just don't call `node.load_configuration()` if you want to work with the values from the OD.  They are automatically read in when initializing the node's OD from a DCF file (or defaults from the EDS).  The only missing step would be to update the PDO configuration as well, which `node.pdo.read(from_od=True)` does.

**Comment by sveinse** [comment]
The PDO state is independent of the OD state and when the `RemoteNode()` class instance is created. The PDO objects are created and referenced to the OD, but their mapping state vars, like `.cob_id`, `.rtr_allowed` etc. are only populated with a call to `PdoMap.read()`. So the PDO objects are not useable out of box. `RemoteNode(..., load_od=True)` can be used but it ends up doing remote SDO-requests since `PdoMap.read()` has default `from_od=False`.

Maybe `PdoMaps().__init__()` when the `PdoMap()` instance is created from the OD should also call `PdoMap.read(from_od=True)` to sync the PDO state with the (default) OD state? This way the PDOs would ready to use after creating the `RemoteNode` instance. If the user needs to update _from_ the remote PDO configuration, the `PdoMap.read(from_od=False)` will do the job as per usual.

**Comment by erlend-aasland** [comment]
André:
> The `.load_configuration()` method is more equivalent to `pdo.write()`, though. It goes through all objects in the OD actually and downloads the values to the remote node via SDO. Then to avoid having mismatching information about the PDO-related objects, it reads back (uploads) the PDO records again via SDO.

It would be nice to add hint about this in the `.load_configuration` docs.

**Comment by erlend-aasland** [comment]
> Maybe `PdoMaps().__init__()` when the `PdoMap()` instance is created from the OD should also call `PdoMap.read(from_od=True)` to sync the PDO state with the (default) OD state? This way the PDOs would ready to use after creating the `RemoteNode` instance. If the user needs to update _from_ the remote PDO configuration, the `PdoMap.read(from_od=False)` will do the job as per usual.

First: I think this discussion belongs on the tracker, not on a PR.
Second: I prefer APIs without side-effects, especially when instantiating objects. IOW, when I create a `RemoteNode` instance, I would not want it to default to doing I/O-operations; I prefer to control those explicitly, since it makes for code that is easier to reason about.

**Comment by sveinse** [comment]
> Second: I prefer APIs without side-effects, especially when instantiating objects. IOW, when I create a `RemoteNode` instance, I would not want it to default to doing I/O-operations; I prefer to control those explicitly, since it makes for code that is easier to reason about.

It doesn't do IO unless `RemoteNode(..., load_od=True)` is used.

**Comment by erlend-aasland** [comment]
FTR, I'd like to take a stab at improving the docs; it is a good opportunity for me to get to know the various APIs of this excellent library better.

Cookie-licking disclaimer: if I don't post a PR within a day or two, feel free to beat me to it :)

**Pull request closed without merging**